mate-panel/panel-run-dialog.c

```diff
@@ -766,79 +776,83 @@ static GSList *
 				 GSList         *list)
 {
-	MateMenuTreeItem *aliased_item;
-
-	aliased_item = matemenu_tree_alias_get_item (alias);
-
-	switch (matemenu_tree_item_get_type (aliased_item)) {
-	case MATEMENU_TREE_ITEM_ENTRY:
-		list = g_slist_append (list, matemenu_tree_item_ref (aliased_item));
-		break;
-
-	case MATEMENU_TREE_ITEM_DIRECTORY:
-		list = get_all_applications_from_dir (MATEMENU_TREE_DIRECTORY (aliased_item),
-						      list);
-		break;
-
-	default:
-		break;
+	gpointer item;
+	switch (matemenu_tree_alias_get_aliased_item_type (alias)) {
+		case MATEMENU_TREE_ITEM_ENTRY:
+			item = matemenu_tree_alias_get_aliased_entry (alias);
+			list = g_slist_append (list, (MateMenuTreeEntry*) item);
+			break;
+		case MATEMENU_TREE_ITEM_DIRECTORY:
+			item = matemenu_tree_alias_get_aliased_directory (alias);
+			list = get_all_applications_from_dir ((MateMenuTreeDirectory*) item, list);
+			break;
+		default:
+			break;
 	}
-
-	matemenu_tree_item_unref (aliased_item);
-
 	return list;
 }
 
 static GSList *
 get_all_applications_from_dir (MateMenuTreeDirectory *directory,
 			       GSList             *list)
 {
-	GSList *items;
-	GSList *l;
-
-	items = matemenu_tree_directory_get_contents (directory);
-
-	for (l = items; l; l = l->next) {
-		switch (matemenu_tree_item_get_type (l->data)) {
-		case MATEMENU_TREE_ITEM_ENTRY:
-			list = g_slist_append (list, matemenu_tree_item_ref (l->data));
-			break;
-
-		case MATEMENU_TREE_ITEM_DIRECTORY:
-			list = get_all_applications_from_dir (l->data, list);
-			break;
-
-		case MATEMENU_TREE_ITEM_ALIAS:
-			list = get_all_applications_from_alias (l->data, list);
-			break;
-
-		default:
-			break;
+	MateMenuTreeIter *iter;
+	MateMenuTreeItemType type;
+
+	iter = matemenu_tree_directory_iter (directory);
+	while ((type = matemenu_tree_iter_next (iter)) != MATEMENU_TREE_ITEM_INVALID) {
+		gpointer item;
+		switch (type) {
+			case MATEMENU_TREE_ITEM_ENTRY:
+				item = matemenu_tree_iter_get_entry (iter);
+				list = g_slist_append (list, (MateMenuTreeEntry*) item);
+				break;
+
+			case MATEMENU_TREE_ITEM_DIRECTORY:
+				item = matemenu_tree_iter_get_directory (iter);
+				list = get_all_applications_from_dir ((MateMenuTreeDirectory*)item, list);
+				break;
+
+			case MATEMENU_TREE_ITEM_ALIAS:
+				item = matemenu_tree_iter_get_alias (iter);
+				list = get_all_applications_from_alias ((MateMenuTreeAlias *)item, list);
+				break;
+
+			default:
+				g_assert_not_reached ();
+				break;
 		}
-
-		matemenu_tree_item_unref (l->data);
 	}
-
-	g_slist_free (items);
-
+	matemenu_tree_iter_unref (iter);
 	return list;
 }
 
 static GSList* get_all_applications(void)
 {
 	MateMenuTree* tree;
 	MateMenuTreeDirectory* root;
+	GError        *error = NULL;
 	GSList* retval;
 
-	tree = matemenu_tree_lookup("mate-applications.menu", MATEMENU_TREE_FLAGS_NONE);
-	matemenu_tree_set_sort_key(tree, MATEMENU_TREE_SORT_DISPLAY_NAME);
+	tree = matemenu_tree_new ("mate-applications.menu", MATEMENU_TREE_FLAGS_NONE);
```

Comment:
Looks like `MATEMENU_TREE_SORT_DISPLAY_NAME` is lost here.

Reply:
ok, I will add it.